fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/PrimaryAccessHint.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * PrimaryAccessHint.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.cascades;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Represents reading a table directly without using an index plus fetch.
+ */
+public class PrimaryAccessHint implements AccessHint {
+
+    public PrimaryAccessHint() {
```

Comment:
Hmmm -- I am pretty sure Sonar doesn't like this. Please remove the constructor entirely.